### DISCUSSION TITLE
Disable {Ctrl,Shift}+Enter opening in new {tab,window} in insert mode.

### DISCUSSION THREAD

**Discussion by qsmodo** [qutebrowser/qutebrowser#6242] [Q&A] [answered]
In insert mode, {Ctrl,Shift}+Enter can open a query in new a {tab,window}. To reproduce it, go to [Duckduckgo](https://ddg.gg), type something and press Shift+Enter.

The issue is that many times I am not fast enough to release Shift before pressing Enter and then it is a micro-frustration to see the query opening in a whole new window.

I have tried

```
config.bind('<Shift-Enter>','<Enter>',mode='insert')
```

but it had no effect, and I didn't alter the default key mappings that would seem to do just what I wanted, namely to make Shift+Enter and Ctrl+Enter do the same as a plain Enter:

```
{..., "<Shift-Enter>": "<Return>", "<Shift-Return>": "<Return>"}
```

BTW thanks for the excellent browser :)

**Comment by The-Compiler** [comment, marked as answer]
There are various issues at play here:

- The keybinding you mention is handled by QtWebEngine (or the underlying Chromium), not by qutebrowser itself
- However, `bindings.key_mappings` only affects qutebrowser bindings
- Qt calls the key "Return" ("Enter" is the numpad enter), and while "Enter" is mapped to "Return", "Shift-Enter" isn't mapped to "Shift-Return".

Thus, if you change your binding to `<Shift-Return>` instead, you will see that it "works" - however, it shows an error because keys are bound to commands in qutebrowser, and `<Enter>` isn't a valid command. You can however use the [`:fake-key` command](https://github.com/qutebrowser/qutebrowser/blob/master/doc/help/commands.asciidoc#fake-key) to send a key to the underlying engine.

Thus, this should do what you want:

```python
config.bind('<Shift-Return>', ':fake-key <Return>', mode='insert')
```

> BTW thanks for the excellent browser :)

You're welcome! :slightly_smiling_face:
